### ISSUE TITLE
[Question] Make a formula depend on Xcode 8 only if it's built from sources

### ISSUE THREAD

**Issue by AliSoftware** [Homebrew/homebrew-core#7763]
- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [X] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Context of the Problem

I'm the author of SwiftGen, which is a tool written in Swift and published via Homebrew ([Formula here](https://github.com/Homebrew/homebrew-core/blob/master/Formula/swiftgen.rb))

Because SwiftGen is written in Swift 3, it requires Xcode 8+ to build from source. To address this, I've [added the `depends_on :xcode => "8.0"` line in my Formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/swiftgen.rb).

The problem is that once Homebrew's Jenkins have built the bottles for SwiftGen and people start installing SwiftGen from them, thus getting the precompiled version of SwiftGen instead of having to build it from Source, they still have this requirement, while it's not needed.

### TL;DR

Is it possible to depend on a version of Xcode only for building the formula from source, but discard it when installed from bottle?

#### Only use the Xcode 8 dependency if built from source, not downloaded from bottle?

The consequence of all that being that Homebrew refuses to install the formula if the user has `xcode-select -p` set to Xcode 7… even if, when installed from the bottles, it's not needed to have Xcode 8 installed and `xcode-select`'d

Is there a way to indicate in my formula that th Xcode 8 dependency is only for building SwiftGen from source, but not necessary when downloaded via the bottle?

#### Depend on installed Xcode, not selected Xcode?

Wouldn't it be better if `depends_on :xcode => xcv` would check if Xcode version xcv was *present* on the machine, even if it wasn't the one selected? For example if the user has both Xcode 7 and Xcode 8 on their machine but `xcode-select`'d Xcode 7, it's still possible to build with Xcode 8 by setting the `DEVELOPER_DIR=…` env variable in the shell executing the installation commands.

That's what I actually do in SwiftGen's Rakefile — using `mdfind` to search for all Xcodes present on the user's system with a given version then get its path and use it as a `DEVELOPER_DIR` to execute my commands. This way, it works even if the Xcode I want to use isn't the default selected one.

**Comment by MikeMcQuaid** [comment]
> Is it possible to depend on a version of Xcode only for building the formula from source, but discard it when installed from bottle?

Use `depends_on :xcode => ["8.0", :build]`

**Comment by AliSoftware** [comment]
Ohhhhh ✨ 

@MikeMcQuaid Where can I find that in the documentation? Would it be worth doing a PR to add that info somewhere, or is it already documented and I just missed it?

**Comment by MikeMcQuaid** [comment]
> @MikeMcQuaid Where can I find that in the documentation? Would it be worth doing a PR to add that info somewhere, or is it already documented and I just missed it?

It's probably not explicitly documented for Xcode but adding `:build` to any dependency makes it a build-time only one. Most other formulae use this format for Xcode; generally the best documentation is looking at other formulae.

**Comment by AliSoftware** [comment]
Cool thanks!

Th other question (about using present Xcode instead of selected Xcode) is more about Homebrew itself (and it's implementation on how it resolves and uses `depends_on :xcode`) than a question about formulae, so I guess we can close this issue and resolve that question in the Brew repo?

**Comment by MikeMcQuaid** [comment]
> The other question (about using present Xcode instead of selected Xcode) is more about Homebrew itself (and it's implementation on how it resolves and uses depends_on :xcode) than a question about formulae, so I guess we can close this issue and resolve that question in the Brew repo?

Just to save you an issue: I think that's something we'd consider accepting a PR for but wouldn't plan on implementing ourselves.
